zebra/rtadv.c

```diff
@@ -493,8 +493,8 @@ static int rtadv_timer(struct thread *thread)
 
 	RB_FOREACH (vrf, vrf_id_head, &vrfs_by_id)
 		FOR_ALL_INTERFACES (vrf, ifp) {
-			if (if_is_loopback_or_vrf(ifp)
-			    || !if_is_operative(ifp))
+			if (if_is_loopback_or_vrf(ifp) || !if_is_operative(ifp)
+				|| if_lookup_by_index(ifp->ifindex, zvrf->vrf->vrf_id) == NULL)
```

Comment:
Ok, this code should work correctly, but I don't like that it uses undocumented behavior of `if_lookup_by_index` that it doesn't take into account the vrf_id.
How about we make the code more explicit and use the following check instead:
`if_is_loopback_or_vrf(ifp) || !if_is_operative(ifp) || (vrf_is_backend_netns() && ifp->vrf->vrf_id != zvrf->vrf->vrf_id)`?